packages/grid/x-data-grid/src/utils/createSelector.ts

```diff
@@ -48,12 +48,99 @@ const missingInstanceIdWarning = buildWarning([
   'To fix, call it with `apiRef`, e.g. `mySelector(apiRef)`, or pass the instance ID explicitly, e.g `mySelector(state, apiRef.current.instanceId)`.',
 ]);
 
-export const createSelector: CreateSelectorFunction = (...args: any) => {
+function isAPIRef(value: any) {
+  return 'current' in value && 'instanceId' in value.current;
+}
+
+const DEFAULT_INSTANCE_ID = { id: 'default' };
+
+export const createSelector = ((
+  a: Function,
+  b: Function,
+  c?: Function,
+  d?: Function,
+  e?: Function,
+  f?: Function,
+  ...rest: any[]
+) => {
+  if (rest.length > 0) {
+    throw new Error('Unsupported number of selectors');
+  }
+
+  let selector: any;
+
+  if (a && b && c && d && e && f) {
+    selector = (stateOrApiRef: any, instanceIdParam: any) => {
+      const isAPI = isAPIRef(stateOrApiRef);
```

Comment:
Will it look a bit nicer?
```suggestion
      const isApiRef = checkIsApiRef(stateOrApiRef);
```